vortex-serde/src/layouts/read/mod.rs

```diff
@@ -53,56 +56,27 @@ pub enum ReadResult {
     Batch(Array),
 }
 
+#[derive(Debug)]
+pub enum RangeResult {
+    ReadMore(Vec<Message>),
+    Rows(Option<RowSelector>),
+}
+
 pub trait LayoutReader: Debug + Send {
+    /// Produce sets of row ranges to read from underlying layouts.
+    ///
+    /// Empty RangeResult indicates layout is done producing ranges
+    fn next_range(&mut self) -> VortexResult<RangeResult>;
+
     /// Reads the data from the underlying layout
     ///
     /// The layout can either return a batch data, i.e. an Array or ask for more layout messages to
```

Comment:
Does it have to return exactly the RowSelection requested? Or can it return a subset? A prefix?

Reply:
It does which I think is a bad api. I think we have to change the Array here to come with a range so that we can return them out of order or partially. Right now this has to be always batch covering the selection and otherwise it might lead to issues.